app/src/main/java/de/westnordost/streetcomplete/quests/opening_hours_signed/CheckOpeningHoursSigned.kt

```diff
@@ -0,0 +1,115 @@
+package de.westnordost.streetcomplete.quests.opening_hours_signed
+
+import de.westnordost.streetcomplete.data.osm.mapdata.MapDataWithGeometry
+import de.westnordost.osmfeatures.FeatureDictionary
+import de.westnordost.streetcomplete.R
+import de.westnordost.streetcomplete.data.osm.osmquests.Tags
+import de.westnordost.streetcomplete.data.elementfilter.toElementFilterExpression
+import de.westnordost.streetcomplete.data.meta.getLastCheckDateKeys
+import de.westnordost.streetcomplete.data.meta.isKindOfShopExpression
+import de.westnordost.streetcomplete.data.meta.setCheckDateForKey
+import de.westnordost.streetcomplete.data.meta.toCheckDate
+import de.westnordost.streetcomplete.data.meta.updateCheckDateForKey
+import de.westnordost.streetcomplete.data.osm.mapdata.Element
+import de.westnordost.streetcomplete.data.osm.mapdata.filter
+import de.westnordost.streetcomplete.data.osm.osmquests.OsmElementQuestType
+import de.westnordost.streetcomplete.data.user.achievements.QuestTypeAchievement.CITIZEN
+import de.westnordost.streetcomplete.ktx.containsAny
+import de.westnordost.streetcomplete.quests.YesNoQuestAnswerFragment
+import java.time.Instant
+import java.time.LocalDateTime
+import java.time.ZoneId
+import java.util.concurrent.FutureTask
+
+class CheckOpeningHoursSigned (
+    private val featureDictionaryFuture: FutureTask<FeatureDictionary>
+) : OsmElementQuestType<Boolean> {
+
+    private val filter by lazy { """
+        nodes, ways, relations with
+          opening_hours:signed = no
+          and (
+            $hasOldOpeningHoursCheckDateFilter
+            or older today -1 years
+          )
+          and access !~ private|no
+          and (name or brand or noname = yes or name:signed = no or amenity = recycling)
+    """.trimIndent().toElementFilterExpression() }
```

Comment:
I think that `trimIndent` is unnecessary for element filters, as whitespace is ignored anyway?

```suggestion
    """.toElementFilterExpression() }
```